hw/ip/edn/data/edn.hjson

```diff
@@ -0,0 +1,264 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+{ name: "edn",
+  clock_primary: "clk_i",
+  bus_device: "tlul",
+  bus_host: "none",
+  param_list: [
+    { name: "NumEndPoints",
+      desc: "Number of end points to be supported by this block",
+      type: "int",
+      default: "4",
+      local: "true"
+    },
+    { name: "EndPointBusWidth0",
```

Comment:
btw, we don't have an easy way of overriding these parameters per instantiation of EDN :/
That's kind of a limitation on the way our system is cobbled together.  So if you intend for these to be overrideable at instantiation, we may have to consider a different approach.

Reply:
Yikes. Well maybe we make the trade of N number of ports with fixed bus sizes, and let the blocks add their own packer to downsize again. Maybe something like a 32bit bus. If that is considered too wide for chip distribution, then 16 or 8, and peripheral block collect up what they want with a FIFO or equivalent. Please think about this. Once we "decide:, this will also impact verification configuration etc., so would be nice to decide on this point.
Also, for ports not used (assuming a fixed case as you hinted), these can be tied off and we hope that synthesis will wash them away.

Reply:
@eunchan 
what do you think Eunchan? Even with templating we can't really support different values per.
So I think the way to do this would be to pass in parameters from the the top level and into the regfile. 
However I think this would be a pretty big change for the regfile...

Mark i think your suggestion seems reasonable, but it's probably something we should bring-up at the silicon meeting or some other wider forum to see what makes the msot sense.